packages/kbn-i18n/src/angular/filter.test.js

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import angular from 'angular';
+import 'angular-mocks';
+import { i18nProvider } from './provider';
+import { i18nFilter } from './filter';
+import * as i18n from '../core/i18n';
+
+jest.mock('../core/i18n', () => ({
+  translate: jest.fn(),
+}));
+
+angular
+  .module('app', [])
+  .provider('i18n', i18nProvider)
+  .filter('i18n', i18nFilter);
+
+describe('i18nFilter', () => {
+  let filter;
+
+  beforeEach(angular.mock.module('app'));
+  beforeEach(
+    angular.mock.inject(i18nFilter => {
+      filter = i18nFilter;
+    })
+  );
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+
+  it('provides wrapper under i18n translate function', () => {
+    const id = 'id';
+    const defaultMessage = 'default-message';
+    const values = {};
+
+    filter(id, { defaultMessage, values });
+
+    expect(i18n.translate).toHaveBeenCalledTimes(1);
```

Comment:
__issue:__ same note here, can we _not_ mock `../core/i18n` and check what is really generated by i18n?

Reply:
I would not like to do it because of 2 reasons:
1. We use spies and test if function was called and which params were used. We cannot do it with original function.
2. It's not integration test, so I would like to encapsulate logic of `i18n` translate function, as it could be reimplemented and will lead many changes in other files. Otherwise, as in this case, we anyway will check just call and params, not result itself.

Reply:
> We use spies and test if function was called and which params were used. We cannot do it with original function.

If we check that directive sets correct `html/text` we don't have to rely on spies at all. Why do we need them?

> It's not integration test, so I would like to encapsulate logic of i18n translate function, as it could be reimplemented and will lead many changes in other files. 

Right, it's not, you test directive as a unit isolated from external world, `i18n.translate` its internal detail. It's arguable of course, but testing directive with non-mocked `i18n` doesn't feel like integration test to me since `i18n` is a core part of this directive.

> so I would like to encapsulate logic of i18n translate function, as it could be reimplemented and will lead many changes in other files. 

Exactly, `i18n.translate` can even change call signature and this unit test should not required any changes, the only thing we care in this test is what directive produced as the result (`element.html()` or `element.text()`), it doesn't matter how it produced that (calling `i18n.translate` or anything else).

> Otherwise, as in this case, we anyway will check just call and params, not result itself.

Not sure I'm following, what do you mean? What I was suggesting is just verifying `element.html()` or `element.text()` instead of `toHaveBeenCalledWith`. Currently we just test that `i18n` was called and it doesn't mean that content of the element has been updated accordingly :)